### PR TITLE
Basic(ish) Lift stuff

### PR DESCRIPTION
**Pull request by andrewmartin5**

### CODE REVIEW

File: src/main/java/frc/robot/maps/RobotMap.java [hunk 1]
```diff
@@ -63,14 +68,40 @@ public ControlPanelMap getControlPanelMap() {
...
-        public SendableSpeedController elevator() {
+        public SendableSpeedController elevatorLeft() {
+            return new MockSpeedController();
+        }
+
+        public SendableSpeedController elevatorRight() {
             return new MockSpeedController();
         }
+
+        public Solenoid liftBrake() {
```
Comment (bot190) [inline src/main/java/frc/robot/maps/RobotMap.java]: This should be `ISolenoid`

File: src/main/java/frc/robot/maps/RobotMap.java [hunk 2]
```diff
@@ -63,14 +68,40 @@ public ControlPanelMap getControlPanelMap() {
...
+            return new Solenoid(0);
```
Comment (bot190) [inline src/main/java/frc/robot/maps/RobotMap.java]: Then this can be `MockSolenoid`

File: src/main/java/frc/robot/maps/RobotMap.java [hunk 3]
```diff
@@ -63,14 +68,40 @@ public ControlPanelMap getControlPanelMap() {
...
+
+        public InvertDigitalInput upperLiftLimit() {
+            return new InvertDigitalInput(0);
+        }
+
+        public InvertDigitalInput lowerLiftLimit() {
+            return new InvertDigitalInput(0);
+        }
+
+        public Encoder getLeftEncoder() {
```
Comment (bot190) [inline src/main/java/frc/robot/maps/RobotMap.java]: These should be `IEncoder`, and return `MockEncoder`.

File: src/main/java/frc/robot/maps/RobotMap.java [hunk 4]
```diff
@@ -63,14 +68,40 @@ public ControlPanelMap getControlPanelMap() {
...
+
+        public SendableSpeedController elevatorRight() {
             return new MockSpeedController();
         }
+
+        public Solenoid liftBrake() {
+            return new Solenoid(0);
+        }
+
+        public InvertDigitalInput upperLiftLimit() {
```
Comment (bot190) [inline src/main/java/frc/robot/maps/RobotMap.java]: These should return `BooleanSupplier`

File: src/main/java/frc/robot/maps/TempestMap.java [hunk 5]
```diff
@@ -29,4 +29,5 @@ public EncodedSpeedController getLeft() {
             }
         };
     }
+
```
Comment (bot190) [inline src/main/java/frc/robot/maps/TempestMap.java]: Please remove this change.

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 6]
```diff
@@ -60,6 +60,3 @@ public CommandBase drive(DoubleSupplier forward, DoubleSupplier turn) {
         }, this);
     }
 }
-
-
-
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Drive.java]: As well as these changes.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 7]
```diff
@@ -1,36 +1,140 @@
...
+        elevatorMotor = map.elevatorLeft();
+        elevatorBrake = map.liftBrake();
+    }
+
+    // TO DO Confgure encoders to actually read a position
+    int getPosition = liftHeights.Bottom.value();
+
+    // TODO make these heights reflect where we actually want them to be
+    public enum liftHeights {
+        Top(3), Middle(2), Bottom(1);
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Lift.java]: These values should be in the game manual I believe.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 8]
```diff
@@ -1,36 +1,140 @@
...
         super();
-        elevatorMotor = map.elevator();
+        leftEncoder = map.getLeftEncoder();
+        rightEncoder = map.getRightEncoder();
+        elevatorMotor = map.elevatorLeft();
+        elevatorBrake = map.liftBrake();
+    }
+
+    // TO DO Confgure encoders to actually read a position
+    int getPosition = liftHeights.Bottom.value();
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Lift.java]: You really want to be doing this inside of a command  in either the `execute` or `end` function.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 9]
```diff
@@ -1,36 +1,140 @@
...
 public class Lift extends SubsystemBase {
 
+    private Encoder leftEncoder;
+    private Encoder rightEncoder;
     private SendableSpeedController elevatorMotor;
+    private Solenoid elevatorBrake;
 
     private static final double elevatorMotorSpeed = 1;
 
+    boolean isBraked = elevatorBrake.get();
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Lift.java]: Same as `getPosition`, you should get this value regularly inside a command.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 10]
```diff
@@ -1,36 +1,140 @@
...
+
+        // Returning the level the lift is at (top middle or bottom)
+        private liftHeights(int iPosition) {
+            this.iPosition = iPosition;
+        }
+    }
+
+    public CommandBase manualLift(DoubleSupplier speed) {
+        return new RunCommand(() -> {
+            elevatorMotor.set(speed.getAsDouble());
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Lift.java]: This needs to disengage the brake when driving upwards otherwise we'll break things!

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 11]
```diff
@@ -1,36 +1,140 @@
...
+        return new InstantCommand(() -> {
+            if (isBraked == false) {
+                elevatorBrake.set(true);
+            } else
+                elevatorBrake.set(false);
+
+        });
+    }
+
+    public CommandBase extend() {
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Lift.java]: This should also disengage the brake before driving the elevator, then engage it when it stops.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 12]
```diff
@@ -1,36 +1,140 @@
...
         return new StartEndCommand(() -> {
             elevatorMotor.set(-elevatorMotorSpeed);
         }, () -> {
             elevatorMotor.stopMotor();
         }, this);
     }
 
-    public CommandBase down() {
+    public CommandBase retract() {
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Lift.java]: Note that this does not need to do anything with the brake (since it's really a ratchet)

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 13]
```diff
@@ -1,36 +1,140 @@
...
         return new StartEndCommand(() -> {
             elevatorMotor.set(elevatorMotorSpeed);
         }, () -> {
             elevatorMotor.stopMotor();
         }, this);
     }
+
+    public CommandBase goToPoint(liftHeights iPoint) {
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Lift.java]: This should probably be `goToHeight`, though we likely don't need this at all.

File: src/main/java/frc/robot/Robot.java [hunk 14]
```diff
@@ -40,8 +40,8 @@
 public class Robot extends TimedRobot {
 
     private Command autonomousCommand;
-    final private ButtonXboxController driveController = new ButtonXboxController(1);
-    final private ButtonXboxController copilotController = new ButtonXboxController(5);
+    private ButtonXboxController driveController = new ButtonXboxController(1);
+    private ButtonXboxController copilotController = new ButtonXboxController(5);
```
Comment (bot190) [inline src/main/java/frc/robot/Robot.java]: Are these changes necessary? If not, please revert this.

**Review by bot190 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Robot.java [hunk 15]
```diff
@@ -40,8 +42,8 @@
 public class Robot extends TimedRobot {
 
     private Command autonomousCommand;
-    final private ButtonXboxController driveController = new ButtonXboxController(1);
-    final private ButtonXboxController copilotController = new ButtonXboxController(5);
+    final public ButtonXboxController driveController = new ButtonXboxController(1);
+    final public ButtonXboxController copilotController = new ButtonXboxController(5);
```
Comment (msoucy) [inline src/main/java/frc/robot/Robot.java]: :(

Don't make them public. It makes kittens cry. Do you want sad kittens?

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 16]
```diff
@@ -1,36 +1,159 @@
...
+
+        // Returning the level the lift is at (top middle or bottom)
+        private liftHeights(int iPosition) {
+            this.iPosition = iPosition;
+        }
+    }
+
+    public CommandBase moveLift(DoubleSupplier speed) {
+        // The command is named "Move Lift" and requires this subsystem.
+        return new CommandBase() {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Lift.java]: Use a `FunctionalCommand` here instead - and make sure it uses the subsystem!

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 17]
```diff
@@ -1,36 +1,159 @@
...
+
+            }
+
+        };
+    }
+
+    public InstantCommand engageBrake() {
+        return new InstantCommand(() -> {
+            elevatorBrake.set(true);
+        });
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Lift.java]: This command doesn't use the subsystem.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 18]
```diff
@@ -1,36 +1,159 @@
...
-    public CommandBase up() {
-        return new StartEndCommand(() -> {
-            elevatorMotor.set(-elevatorMotorSpeed);
-        }, () -> {
-            elevatorMotor.stopMotor();
-        }, this);
+    public InstantCommand disengageBrake() {
+        return new InstantCommand(() -> {
+            elevatorBrake.set(false);
+        });
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Lift.java]: This command doesn't use the subsystem.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 19]
```diff
@@ -1,36 +1,159 @@
...
 
-    public CommandBase down() {
-        return new StartEndCommand(() -> {
-            elevatorMotor.set(elevatorMotorSpeed);
-        }, () -> {
-            elevatorMotor.stopMotor();
-        }, this);
+    public InstantCommand toggleBrake() {
+        return new InstantCommand(() -> {
+            if (isBraked == false) {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Lift.java]: - This command doesn't use the subsystem.
- Use `elevatorBrake.set(!isBraked)`
- Make sure you update `elevatorBrake` here and elsewhere to keep the value accurate.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 20]
```diff
@@ -1,36 +1,159 @@
...
+                elevatorBrake.set(true);
+            } else
+                elevatorBrake.set(false);
+
+        });
+    }
+
+    public CommandBase goToHeight(liftHeights iPoint) {
+
+        return new CommandBase() {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Lift.java]: Since this doesn't have state, use a `FunctionalCommand`

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 21]
```diff
@@ -1,36 +1,159 @@
...
+
+import com.chopshop166.chopshoplib.outputs.ISolenoid;
+import com.chopshop166.chopshoplib.outputs.PIDSpeedController;
 import com.chopshop166.chopshoplib.outputs.SendableSpeedController;
+import com.chopshop166.chopshoplib.sensors.IEncoder;
+import com.revrobotics.CANPIDController;
 
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj.GenericHID.Hand;
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Lift.java]: Clean up your imports.

**Review by msoucy — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 22]
```diff
@@ -72,7 +98,7 @@ public CommandBase moveLift(DoubleSupplier speed) {
             elevatorBrake.set(false);
             isBraked = false;
         }, () -> {
-            elevatorMotor.set(speed.getAsDouble());
+            liftSpeed(speed.getAsDouble());
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Lift.java:90]: If you're using `liftSpeed` here, you may not need the initialize and end functions to do anything other than set it to 0.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 23]
```diff
@@ -106,12 +132,12 @@ public CommandBase goToHeight(liftHeights iPoint) {
             elevatorBrake.set(false);
         }, () -> {
             if (getPosition > 2) {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Lift.java]: Where does 2 come from?

**Review by msoucy — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/maps/FrancoisMap.java [hunk 24]
```diff
@@ -88,6 +97,38 @@ public PIDSparkMax elevator() {
...
+            @Override
+            public BooleanSupplier lowerLiftLimit() {
+                // TODO Change this to a real Channel
+                InvertDigitalInput lowerLimit = new InvertDigitalInput(1);
+
+                return lowerLimit::get;
+            }
+
+            @Override
+            public IEncoder getLiftEncoder() {
```
Comment (bot190) [inline src/main/java/frc/robot/maps/FrancoisMap.java:168]: This encoder is really coming from the `CANSparkMax`. We probably need to define `leader` outside `elevator()` and then return the `SparkMaxEncoder` object here. Though we may not actually need the separate encoder at all if we get the PID working correctly.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 25]
```diff
@@ -1,36 +1,146 @@
...
+        private double minLiftHeight = 51;
+
+        // Returning an double to compare whether we're in the right place or not
+        public double value() {
+            return this.iPosition - minLiftHeight;
+        }
+
+        // Returning the level the lift is at (top middle or bottom)
+        private liftHeights(double iPosition) {
+            this.iPosition = iPosition;
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Lift.java]: This seems highly unlikely to ever be true. I suggest adding some sort of range around `this.iPosition`.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 26]
```diff
@@ -1,36 +1,146 @@
...
+        elevatorBrake.set(isBraked);
+        elevatorMotor.set(speed);
+
+    }
+
+    // TODO make these heights reflect where we actually want them to be
+    public enum liftHeights {
+        Top(52.25), Middle(63), Bottom(78.125);
+
+        private double iPosition;
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Lift.java]: Why is this `iPosition`? Maybe `mPosition`

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 27]
```diff
@@ -1,36 +1,146 @@
...
+            double getPosition = liftEncoder.getDistance();
+            if (getPosition > iPoint.value()) {
+                liftSpeed(elevatorMotorSpeed);
+            } else {
+                liftSpeed(-elevatorMotorSpeed);
+            }
+        }, (Boolean interrupted) -> {
+            liftSpeed(0);
+            elevatorBrake.set(true);
+            isBraked = true;
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Lift.java]: These aren't necessary since `liftSpeed` does this.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 28]
```diff
@@ -1,36 +1,146 @@
...
+            liftSpeed(speed.getAsDouble());
+        }, (Boolean interrupted) -> {
+            liftSpeed(0);
         }, () -> {
-            elevatorMotor.stopMotor();
+            return false;
+        }, this);
+    }
+
+    public InstantCommand engageBrake() {
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Lift.java]: We can probably get rid of `engageBrake` and `disengageBrake`. It might be worth keeping `toggleBrake` for now.

File: src/main/java/frc/robot/subsystems/Lift.java (hunk 9 above)
Comment (bot190) [reply src/main/java/frc/robot/subsystems/Lift.java]: Instead of defining a variable here, we could just call `elevatorBrake.get()` in the same places. Tracking the variable is more error prone.

**Review by bot190 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Drive.java (hunk 6 above)
Comment (andrewmartin5) [reply src/main/java/frc/robot/subsystems/Drive.java]: My visual studio autoformats this out as soon as I save it

File: src/main/java/frc/robot/subsystems/Lift.java (hunk 25 above)
Comment (andrewmartin5) [reply src/main/java/frc/robot/subsystems/Lift.java]: getPosition >= iPoint.value() - 5 && getPosition <= iPoint.value() + 5

This is the best way i have found but it looks really bad. is there a better way to do it?

File: src/main/java/frc/robot/maps/FrancoisMap.java (hunk 24 above)
Comment (andrewmartin5) [reply src/main/java/frc/robot/maps/FrancoisMap.java:168]: wait what?

File: src/main/java/frc/robot/Robot.java [hunk 29]
```diff
@@ -7,6 +7,8 @@
 
 package frc.robot;
 
+import java.util.function.DoubleSupplier;
```
Comment (bot190) [inline src/main/java/frc/robot/Robot.java]: Please remove this extra import.

File: src/main/java/frc/robot/maps/FrancoisMap.java (hunk 24 above)
Comment (bot190) [reply src/main/java/frc/robot/maps/FrancoisMap.java:168]: The `TODO` can be removed and you can skip the intermediate variable.

File: src/main/java/frc/robot/maps/RobotMap.java [hunk 30]
```diff
@@ -63,14 +72,108 @@ public ControlPanelMap getControlPanelMap() {
...
+                }
+            };
+        }
+
+        public ISolenoid liftBrake() {
+            return new MockSolenoid();
         }
+
+        public BooleanSupplier upperLiftLimit() {
+            return () -> false;
```
Comment (bot190) [inline src/main/java/frc/robot/maps/RobotMap.java]: Please use `MockDigitalInput` instead of hard-coding this to `false`.

File: src/main/java/frc/robot/maps/RobotMap.java [hunk 31]
```diff
@@ -63,14 +72,108 @@ public ControlPanelMap getControlPanelMap() {
...
+        }
+
+        public BooleanSupplier lowerLiftLimit() {
```
Comment (bot190) [inline src/main/java/frc/robot/maps/RobotMap.java]: It turns out we aren't going to have a lower limit switch. This can be removed.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 32]
```diff
@@ -1,36 +1,128 @@
...
-    private SendableSpeedController elevatorMotor;
+public class Lift extends SubsystemBase {
 
+    private PIDSpeedController elevatorMotor;
+    private ISolenoid elevatorBrake;
+    private IEncoder liftEncoder;
+    private BooleanSupplier upperLimitSwitch;
+    private BooleanSupplier lowerLimitSwitch;
     private static final double elevatorMotorSpeed = 1;
+    private static final double TOLERANCE_RANGE = 5;
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Lift.java]: What unit is this in? Please put the unit in the name (eg `TOLERANCE_RANGE_INCHES`).

Also, assuming inches 5 is way to large, this should probably be 0.5 or so.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 33]
```diff
@@ -1,36 +1,128 @@
...
+        } else if (speed <= 0) {
+            elevatorBrake.set(true);
+            if (lowerLimitSwitch.getAsBoolean()) {
+                speed = 0;
+            }
+        }
+        elevatorMotor.set(speed);
+    }
+
+    // TODO make these heights reflect where we actually want them to be
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Lift.java]: This is solved right?

**Review by bot190 — CHANGES_REQUESTED** [review]
Just a few more small things for now.

**Comment by msoucy** [comment]
Merge master into this branch, solve the merge conflicts, and then solve Ben's issues, and this should be good.

File: src/main/java/frc/robot/maps/FrancoisMap.java [hunk 34]
```diff
@@ -1,27 +1,36 @@
...
 
+import edu.wpi.first.wpilibj.Solenoid;
+
 @RobotMapFor("Francois")
 public class FrancoisMap extends RobotMap {
 
     @Override
     public DifferentialDriveMap getDriveMap() {
         return new DifferentialDriveMap() {
+            CANSparkMax follower = new CANSparkMax(23, MotorType.kBrushless);
```
Comment (bot190) [inline src/main/java/frc/robot/maps/FrancoisMap.java]: This shouldn't have been moved.

File: src/main/java/frc/robot/maps/FrancoisMap.java [hunk 35]
```diff
@@ -56,10 +64,11 @@ public WDSolenoid deployIntake() {
     @Override
     public ShooterMap getShooterMap() {
         return new ShooterMap() {
+            CANSparkMax follower = new CANSparkMax(26, MotorType.kBrushless);
```
Comment (bot190) [inline src/main/java/frc/robot/maps/FrancoisMap.java]: Nor should this.

File: src/main/java/frc/robot/maps/FrancoisMap.java [hunk 36]
```diff
@@ -80,14 +89,36 @@ public SendableSpeedController spinner() {
     @Override
     public LiftMap getLiftMap() {
         return new LiftMap() {
+            CANSparkMax follower = new CANSparkMax(28, MotorType.kBrushless);
+            CANSparkMax leader = new CANSparkMax(27, MotorType.kBrushless);
+            InvertDigitalInput upperLimit = new InvertDigitalInput(0);
```
Comment (bot190) [inline src/main/java/frc/robot/maps/FrancoisMap.java:145]: This can be defined in `upperLiftLimit()`

File: src/main/java/frc/robot/maps/FrancoisMap.java [hunk 37]
```diff
@@ -80,14 +89,36 @@ public SendableSpeedController spinner() {
     @Override
     public LiftMap getLiftMap() {
         return new LiftMap() {
+            CANSparkMax follower = new CANSparkMax(28, MotorType.kBrushless);
```
Comment (bot190) [inline src/main/java/frc/robot/maps/FrancoisMap.java]: This could be defined in `elevator()`. Only `leader` needed to be moved here.

File: src/main/java/frc/robot/maps/RobotMap.java [hunk 38]
```diff
@@ -63,14 +72,102 @@ public ControlPanelMap getControlPanelMap() {
         return new ControlPanelMap();
     }
 
+    // TODO replace these values once true ports are determined
```
Comment (bot190) [inline src/main/java/frc/robot/maps/RobotMap.java]: Please remove this. There are only mocks in here.

File: src/main/java/frc/robot/maps/RobotMap.java [hunk 39]
```diff
@@ -63,14 +72,102 @@ public ControlPanelMap getControlPanelMap() {
...
+
     public static class LiftMap {
-        public SendableSpeedController elevator() {
-            return new MockSpeedController();
+
+        public PIDSpeedController elevator() {
+            return new PIDSpeedController() {
```
Comment (bot190) [inline src/main/java/frc/robot/maps/RobotMap.java]: This can now become `MockPIDSpeedController`

**Review by bot190 — CHANGES_REQUESTED** [review]
More comments

File: src/main/java/frc/robot/maps/FrancoisMap.java (hunk 37 above)
Comment (msoucy) [reply src/main/java/frc/robot/maps/FrancoisMap.java]: It gives complaints about variables getting closed if they don't do this

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 40]
```diff
@@ -1,36 +1,128 @@
...
+            if (lowerLimitSwitch.getAsBoolean()) {
+                speed = 0;
+            }
+        }
+        elevatorMotor.set(speed);
+    }
+
+    // Heights of the bar at max, min, and middle
+    public enum liftHeights {
+        Top(52.25), Middle(63), Bottom(78.125);
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Lift.java]: I'm confused - why is top lower than bottom?

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 41]
```diff
@@ -1,36 +1,128 @@
...
+        private liftHeights(double mPosition) {
+            this.mPosition = mPosition;
+        }
     }
 
-    public CommandBase up() {
-        return new StartEndCommand(() -> {
-            elevatorMotor.set(-elevatorMotorSpeed);
+    public CommandBase moveLift(DoubleSupplier speed) {
+        // The command is named "Move Lift" and requires this subsystem.
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Lift.java]: This command ain't named nothing anymore!

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 42]
```diff
@@ -1,36 +1,128 @@
...
+        return new FunctionalCommand(() -> {
+            liftSpeed(0);
+        }, () -> {
+            liftSpeed(speed.getAsDouble());
+        }, (Boolean interrupted) -> {
+            liftSpeed(0);
         }, () -> {
-            elevatorMotor.stopMotor();
+            return false;
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Lift.java]: Fun fact: `() -> { return false; }` can be shortened to `() -> false`. Nice little (optional) shorthand.

**Review by msoucy — CHANGES_REQUESTED** [review]

**Review by bot190 — APPROVED** [review]

**Review by msoucy — APPROVED** [review]
Good enough for now
